HIRS_ProvisionerTPM2/src/tpm_version.cpp

```diff
@@ -1,11 +1,125 @@
+#include <sapi/tpm20.h>
+#include <tcti/tcti-tabrmd.h>
+
+#include <arpa/inet.h>
 #include <iostream>
+#include <memory>
+#include <string>
 
 using std::cout;
 using std::endl;
+using std::shared_ptr;
+using std::string;
+
+class SapiContext {
+    TSS2_TCTI_CONTEXT *tctiContext = nullptr;
+    TSS2_SYS_CONTEXT *sapiContext = nullptr;
+
+    SapiContext(TSS2_TCTI_CONTEXT* tcti_ctx, TSS2_SYS_CONTEXT* sapi_ctx)
+            : tctiContext(tcti_ctx), sapiContext(sapi_ctx) {}
+public:
+
+    ~SapiContext() {free(sapiContext); free(tctiContext);}
+    static shared_ptr<SapiContext> create() {
```

Comment:
Can you add some documentation and just some general commenting to this file?

Reply:
will do